diff --git a/src/include/aligned_alloc.h b/src/include/aligned_alloc.h
--- a/src/include/aligned_alloc.h
+++ b/src/include/aligned_alloc.h
@@ -33,5 +33,7 @@ inline _LIBCPP_HIDE_FROM_ABI void* __libcpp_aligned_alloc(std::size_t __alignmen
 // Android only provides aligned_alloc when targeting API 28 or higher.
-#  elif !defined(__ANDROID__) || __ANDROID_API__ >= 28
+// Apple only provides aligned_alloc when targeting macOS 10.15 or higher.
+#  elif (!defined(__ANDROID__) || __ANDROID_API__ >= 28) && \
+      (!defined(__APPLE__) || __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101500)
   // aligned_alloc() requires that __size is a multiple of __alignment,
   // but for C++ [new.delete.general], only states "if the value of an
   // alignment argument passed to any of these functions is not a valid
